com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/utils/FileSystemService.java

```diff
@@ -73,4 +74,15 @@
    */
   @Nonnull
   String decodeURI(@Nonnull String uri);
+
+  /**
+   * This method validate an URI to verify if is defined as extended document, accordining to a
+   * regex.
+   *
+   * @param uri document URI opened in the client
+   * @return true if the document is an extended document, false otherwise
+   */
+  static boolean isFileUnderExtendedSourceFolder(String uri) {
+    return Pattern.matches("file:///(?:(.*?/\\.c4z/\\.extsrcs/.+)|\\.c4z/\\.extsrcs/.+)", uri);
```

Comment:
It is not clear what you mean here. Because (?:<something>) is a non-captured group, but inside you have a captured one. Probably `file:///.*?/\.c4z/\.extsrcs/.+` will be enough.

Reply:
To plan further a refactoring getting rid of ```FileSystemService``` interface (tbd), I will move the method into the ```WorkspaceFileService``` class

Reply:
Why do you want to get rid of it?

Reply:
because we came up that ```WorkspaceFileService``` could be an utility class with no needs of an interface 😓

Reply:
But the initial idea of that was to simplify testing of `CopybookService`. Please, have a look at `CopybookServiceTest`

Reply:
yes and in future you could mock directly the utility class rather than the interface, right? BTW is not a decision that we have to make now! ;)